app/src/main/java/protect/card_locker/ShortcutHelper.java

```diff
@@ -106,6 +119,22 @@ static void removeShortcut(Context context, int cardId) {
         ShortcutManagerCompat.setDynamicShortcuts(context, list);
     }
 
+    static @NotNull Bitmap createAdaptiveBitmap(@NotNull Bitmap in, int paddingColor){
+        int[] basePixels = new int[ADAPTIVE_BITMAP_SIZE * ADAPTIVE_BITMAP_SIZE];
+        paddingColor = ColorUtils.compositeColors(PADDING_COLOR_OVERLAY, paddingColor);
+        Arrays.fill(basePixels, paddingColor);
+        Bitmap ret = Bitmap.createBitmap(ADAPTIVE_BITMAP_SIZE, ADAPTIVE_BITMAP_SIZE, Bitmap.Config.ARGB_8888);
+        ret.setPixels(basePixels, 0, ADAPTIVE_BITMAP_SIZE, 0, 0, ADAPTIVE_BITMAP_SIZE, ADAPTIVE_BITMAP_SIZE);
+        Bitmap resized = Utils.resizeBitmap(in, ADAPTIVE_BITMAP_IMAGE_SIZE);
+        int[] inputPixels = new int[resized.getHeight() * resized.getWidth()];
+        resized.getPixels(inputPixels, 0, resized.getWidth(), 0, 0, resized.getWidth(), resized.getHeight());
+        for(int i = 0;i < inputPixels.length; i++) {
+            inputPixels[i] = ColorUtils.compositeColors(inputPixels[i], paddingColor);
+        }
+        ret.setPixels(inputPixels, 0, resized.getWidth(), (ADAPTIVE_BITMAP_SIZE - resized.getWidth()) / 2, (ADAPTIVE_BITMAP_SIZE - resized.getHeight()) / 2, resized.getWidth(), resized.getHeight());
+        return ret;
+    }
+
```

Comment:
This works but wouldn't it be much simpler to just use a Canvas?

```
Bitmap bitmapWithBackground = Bitmap.createBitmap(in.getWidth(), in.getHeight(), in.getConfig());
Canvas canvas = new Canvas(bitmapWithBackground);
canvas.drawColor(paddingColor);
canvas.drawBitmap(in, 0, 0, null);

return bitmapWithBackground;
```

https://stackoverflow.com/a/4573564